skrifa/src/outline/autohint/instance.rs

```diff
@@ -30,3 +42,134 @@ impl GlyphStyles {
         }
     }
 }
+
+#[derive(Clone)]
+pub(crate) struct Instance {
+    styles: GlyphStyles,
+    metrics: UnscaledStyleMetricsSet,
+    target: Target,
+    is_fixed_width: bool,
+    style: Style,
+}
+
+impl Instance {
+    pub fn new(
+        font: &FontRef,
+        outlines: &OutlineGlyphCollection,
+        coords: &[F2Dot14],
+        target: Target,
+        styles: Option<GlyphStyles>,
+        lazy_metrics: bool,
+    ) -> Self {
+        let styles = styles.unwrap_or_else(|| GlyphStyles::new(outlines));
+        #[cfg(feature = "std")]
+        let metrics = if lazy_metrics {
+            UnscaledStyleMetricsSet::lazy(&styles.0)
+        } else {
+            UnscaledStyleMetricsSet::precomputed(font, coords, &styles.0)
+        };
+        #[cfg(not(feature = "std"))]
+        let metrics = UnscaledStyleMetricsSet::precomputed(font, coords, &styles.0);
+        let is_fixed_width = font
+            .post()
+            .map(|post| post.is_fixed_pitch() != 0)
+            .unwrap_or_default();
+        let style = font.attributes().style;
+        Self {
+            styles,
+            metrics,
+            target,
+            is_fixed_width,
+            style,
+        }
+    }
+
+    pub fn draw(
+        &self,
+        size: Size,
+        coords: &[F2Dot14],
+        glyph: &OutlineGlyph,
+        path_style: PathStyle,
+        pen: &mut impl OutlinePen,
+    ) -> Result<AdjustedMetrics, DrawError> {
```

Comment:
I thought `HintingOptions`, `Target::Smooth` etc. were already exposed, so the subject of the PR means that the hinted metrics etc. would start to be returned? Otherwise I might be missing what is new public API in this PR. I was trying to look at it from the perspective of whether it matches the autohinting preview PR, which I assume it does.

Reply:
A better title for this PR might be “Actually run the autohinter when requested” since the public API already exposed the option but the underlying code ignored it :) There shouldn’t be any changes wrt use from the preview branch.

Reply:
Ack, makes sense now.